README.md

```diff
@@ -22,19 +22,51 @@ modules: {
 
 The user will need a valid Google API key for the suggestions displayed by Google to work as stated in Google Maps Platform documentation: [https://developers.google.com/maps/documentation/javascript/places-autocomplete](https://developers.google.com/maps/documentation/javascript/places-autocomplete). You must connect a billing account to the Google API console project.
 
-## Advanced settings
+Default rendering:
 
-By default, the response from Google will be split into 3 fields: street address, route and city. If you don't need this, you can uncheck the boxes in "Advanced settings / Split Address".
+![](./assets/basic.gif)
 
-Split address demo:
+When submitted, the form's output will contain the input's name and value.
+For example, if the input's name is `address`, the above example would output:
 
-![](./assets/split.gif)
+```js
+{
+  address: 'Tour Eiffel, Avenue Anatole France, Paris, France'
+}
+```
 
-Basic demo:
 
-![](./assets/basic.gif)
+## Advanced settings
 
-Other settings can restrict the number of results
+The number of results can be restricted
 - to a type (addresses, regions, establishments, ...)
 - to a certain area with coordinates
 - to countries (5 at max due to Google policy)
+
+The address can also be split into multiple fields, as allowed by Google. The current list of choices is the following:
+
+![](./assets/splitaddress.png)
+
+Any number of fields can be added and the fields' order can be rearranged.
+
+Rendering example of the above configuration:
+
+![](./assets/split.gif)
+
+When submitted, the form's output will contain the fields names and values (not the input's name).
+For example, the above example would output:
+
+```js
+{
+  street_number: '5',
```

Comment:
Wait, how does this account for the fact that there can be more than one address field included in a form? These names, like "country", seem like they would conflict in any representation that doesn't allow two columns with the same name. They should be based on the name of the address field. Also the complete address field should still be saved.

Let me try to clarify more.

If the address fields present in the form are named "shipping_address" and "billing_address" via the "name" field of the widget, and split fields are activated, I would expect to see this:

```
shipping_address: "5000 Rittenhouse Street, Philadelphia, PA 19144 United States"
shipping_address_street_number: "5810"
shipping_address_route: "Rittenhouse Street"
shipping_address_locality: "Philadelphia"
shipping_address_administrative_area_level_1: "Pennsylvania"
shipping_address_postal_code: "19147"
shipping_address_country: "United States"
```

This then allows for non-conflicting fields for "billing_address".

Does this clarify what I am asking about?

Reply:
Yes, thanks for the clarification.